MetaOptimize.Cli/PIFORunner.cs

```
@@ -0,0 +1,145 @@
+
+            for (int pid = 0; pid < numPackets; pid++)
+            {
+                if (optimalSol.Admit[pid] == 1)
+                {
+                    orderToRankOpt[optimalSol.Order[pid]] = optimalSol.Ranks[pid];
+                }
+
+                if (heuristicSol.Admit[pid] == 1)
+                {
+                    orderToRankHeu[heuristicSol.Order[pid]] = heuristicSol.Ranks[pid];
+                }
+            }
+
+            int numInvOpt = 0;
+            int numInvHeu = 0;
+
+            for (int pid = 0; pid < numPackets; pid++)
+            {
+                numInvOpt += ComputeInversionNum(optimalSol, orderToRankOpt, pid);
+                numInvHeu += ComputeInversionNum(heuristicSol, orderToRankHeu, pid);
+            }
+
+            return (numInvOpt, numInvHeu);
+        }
+
+        /// <summary>
+        /// Runs Bin Packing optimization.
+        /// Uses MainVBP logic (more complete than vbMain).
+        /// </summary>
+        public static void Run(CliArgs args)
+        {
+            var maxRank = args.GetInt("--maxRank", 8);
+            var numPackets = args.GetInt("--numPackets", 18);
+            var numQueues = args.GetInt("--numQueues", 4);
+            var maxQueueSize = args.GetInt("--maxQueueSize", 12);
+            var windowSize = args.GetInt("--windowSize", 12);
+            var burstParam = args.GetDouble("--burstParam", 0.1);
+            var timeout = args.GetDouble("--timeout", 1000);
+            var verbose = args.GetBool("--verbose", false);
+
+            Console.WriteLine($"Packets: {numPackets}, Max Rank: {maxRank}, Queues: {numQueues}");
+            Console.WriteLine($"Max Queue Size: {maxQueueSize}, Window Size: {windowSize}");
+
+            var solver = new GurobiSOS(verbose: Convert.ToInt32(verbose), timeout: timeout);
+
+            // Create encoders - comparing SP-PIFO with drop vs AIFO
+            var h1 = new SPPIFOWithDropAvgDelayEncoder<GRBVar, GRBModel>(
```

Can we take h1 and h2 as command line inputs (PIFOMethod1, PIFOMethod2)? They can be one of the following:
- PIFO
- SPPIFO
- AIFO
- ModifiedSPPIFO

We should also add another command line option ("--ConsiderPktDrop true/false"). Then, depending on whether this is enabled or not, we should call:
- SPPIFOAvgDelayEncoder vs SPPIFOWithDropAvgDelayEncoder
- PIFOAvgDelayOptimalEncoder vs PIFOWithDropAvgDelayEncoder

For AIFO, we only have one encoder "AIFOAvgDelayEncoder", which should be used only when ConsiderPktDrop is true. Otherwise, we should return an exception with something similar to the following message "AIFO only works on shallow buffers and decides whether to admit or drop the packet. As a result, it does apply to cases where we do not want packet drop."

For Modified-SP-PIFO, we also have one encoder "ModifiedSPPIFOAvgDelayEncoder" which should be used only when ConsiderPktDrop is false. Otherwise, we should return an exception ("Do not support packet drop yet"). It also takes two additional inputs: splitQueue and splitRank.

---

done.  part of next check in.